product_supplierinfo_archive/views/product_views.xml

```diff
@@ -21,7 +21,7 @@
         <field name="inherit_id" ref="product.product_supplierinfo_form_view" />
         <field name="arch" type="xml">
             <field name="product_variant_count" position="after">
-                <field name="active" invisible="1" />
```

Comment:
bother, i told you to restore it then you just need to put it back as it was, no need to create more if you haven't tested it that feature. we just need to restore the active field on the form mode with the `invisible="1" `attribute to restore the `Archive` feature on the form view. active is a field used to hide records on the view (soft delete) instead of deleting that record (hard delete) and no one wants to show it on the view :)

read more in: https://www.cybrosys.com/blog/what-are-automatic-and-reserved-fields-in-odoo-17